src/main/java/com/apac/test/apactakehometest/repository/TaxiTripsRepository.java

```diff
@@ -1,9 +1,16 @@
 package com.apac.test.apactakehometest.repository;
 
 import com.apac.test.apactakehometest.model.TaxiTripsModel;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
 import org.springframework.stereotype.Repository;
 
 @Repository
-public interface TaxiTripsRepository extends JpaRepository<TaxiTripsModel, Long>{
+public interface TaxiTripsRepository extends JpaRepository<TaxiTripsModel, Long> {
+    @Query(value = "select tableData from TaxiTripsModel tableData where tableData.taxiID = :taxiID",
+        countQuery = "select count(tableData) from TaxiTripsModel tableData where tableData.taxiID = :taxiID")
+    Page<TaxiTripsModel> searchByTaxiID(@Param("taxiID") Long taxiID, Pageable pageable);
```

Comment:
this is not a search, i believe we have getOne here : https://docs.spring.io/spring-data/jpa/docs/current/api/org/springframework/data/jpa/repository/JpaRepository.html#getOne-ID-

Reply:
@vanduc1102 I already have findById , i think it's similar to getOne ? Just i receiving Optional instead of null if row not found
This method just searching by other field, not included to JPA Repo